tests/unit/Commands/ContinuousIntegration/TravisCi/Config/ConfigWriterTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace QuickStrapUnit\Commands\ContinuousIntegration\TravisCi\Config;
+
+use org\bovigo\vfs\vfsStream;
+use QuickStrap\Commands\ContinuousIntegration\TravisCi\Config\Config;
+use QuickStrap\Commands\ContinuousIntegration\TravisCi\Config\ConfigWriter;
+use Symfony\Component\Yaml\Yaml;
+
+class ConfigWriterTest extends \PHPUnit_Framework_TestCase
+{
+    /** @var  string */
+    private $configFile;
+    
+    /** @var ConfigWriter */
+    private $sut;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        vfsStream::setup('application_path')
+            ->addChild($configFile = vfsStream::newFile('.travis.yml'));
+        
+        $this->configFile = $configFile->url();
+
+        $this->sut = new ConfigWriter();
+    }
+    
+    public function test_can_write_language_to_yml()
+    {
+        $config = new Config('php');
+
+        $actual = $this->sut->toYml($config);
+
+        self::assertEquals("language: php\n", $actual);
+    }
+    
+    public function test_can_write_php_versions_to_yml()
+    {
+        $config = new Config('php');
+        $config->setPhp(['7.0', '5.6']);
+
+        $actual = $this->sut->toYml($config);
+
+        self::assertEquals(<<<YML
+language: php
+php:
+    - '7.0'
+    - '5.6'
+
+YML
+, $actual);
+    }
+    
+    public function test_can_write_before_script_to_yml()
+    {
+        $config = new Config('php');
+        $config->setBeforeScript(['composer self-update', 'composer install']);
+
+        $actual = $this->sut->toYml($config);
+
+        self::assertEquals(<<<YML
+language: php
+before_script:
+    - 'composer self-update'
+    - 'composer install'
+
+YML
+, $actual);
+    }
+    
+    public function test_can_write_script_to_yml()
+    {
+        $config = new Config('php');
+        $config->setScript(['vendor/bin/phpunit', 'vendor/bin/behat']);
+
+        $actual = $this->sut->toYml($config);
+
+        self::assertEquals(<<<YML
+language: php
+script:
+    - vendor/bin/phpunit
+    - vendor/bin/behat
+
+YML
+, $actual);
+    }
+    
+    public function test_can_allow_failures_to_yml()
```

Comment:
these feel like contract and/or collaboration tests to me. on the one hand, i abhor string equivalence tests such as these. they feel very brittle. on the other hand, `toYml()` better adhere to the travis contract, or else things will not work out well.

the thing is... symfony's yaml renders in a very particular way, as you've demonstrated. each build in allow_failures is wrapped with curly brackets. travis/yml is capable of expressing the same information a couple of ways, e.g. without curly brackets. i'd like for tests like these to be more black-box, but the output is not independent of the implementation, and is therefore a little too brittle for my taste. what are your thoughts on this?